src/citrine/resources/table_config.py

```diff
@@ -421,6 +421,10 @@ def __init__(self, *args, team_id: UUID, project_id: UUID = None, session: Sessi
         self.project_id = project_id or args[0]
         self.session: Session = session or args[1]
         self.team_id = team_id
+        if self.project_id is None:
+            raise TypeError("Missing one required argument: project_id.")
+        if self.session is None:
+            raise TypeError("Missing one required argument: session.")
```

Comment:
I noticed we weren't enforcing that these were provided, despite previously being required, and still assumed to be non-`None` in the class.